### DISCUSSION TITLE
Does the `retryExchange` apply to mutations?

### DISCUSSION THREAD

**Discussion by nandorojo** [urql-graphql/urql#2811] [Q&A] [answered]
I don't see this in the docs – do mutations get retried if they fail, or just queries?

**Comment by JoviDeCroock** [comment, marked as answer]
It does yes but you can disable that with `retryIf`

EDIT: at this point I am not sure let me just try in a codesandbox 🤷 
EDIT2: I wasn't going crazy, it does retry it yes

**Reply by nandorojo** [reply]
Got it. How does awaiting the promise work then?

```tsx
const { error } = await executeMutation()

// is it just retrying here without any resolver?
```

**Reply by JoviDeCroock** [reply]
I mean, it's in the exchange chain so it bounces from retry back to fetch and does not deliver the response until it stops retrying

**Reply by nandorojo** [reply]
Oh, perfect. So if it returns `error`, it's already retried the maximum amount of times.
